docs/OBJECT_LOCK_TEST_PLAN.md

```diff
@@ -0,0 +1,157 @@
+the new API actions.
+
+## Functional Tests
+
+### putBucket tests
+
+- passing option to enable object lock updates bucket metadata and enables
+bucket versioning
+
+### putBucketVersioning tests
+
+- suspending versioning on bucket with object lock enabled returns error
+
+### putObject tests
+
+- putting retention configuration on object should be allowed
+- putting invalid retention configuration returns error
+
+### putObjectLockConfiguration tests
+
+- putting configuration as non-bucket-owner user returns AccessDenied error
+- disabling object lock on bucket created with object lock returns error
+- enabling object lock on bucket created without object lock returns
+InvalidBucketState error
+- enabling object lock with token on bucket created without object lock succeeds
+- putting valid object lock configuration when bucket does not have object
+lock enabled returns error (InvalidRequest?)
+- putting valid object lock configuration updates bucket metadata
+- putting invalid object lock configuration returns error
+    - ObjectLockEnabled !== "Enabled"
+    - Rule object doesn't contain DefaultRetention key
+    - Mode !== "GOVERNANCE" or "COMPLIANCE"
+    - Days are not an integer
+    - Years are not an integer
+
+### getObjectLockConfiguration tests
+
+- getting configuration as non-bucket-owner user returns AccessDenied error
+- getting configuration when none is set returns
+ObjectLockConfigurationNotFoundError error
+- getting configuration returns correct object lock configuration for bucket
+
+### putObjectRetention
+
+- putting retention as non-bucket-owner user returns AccessDenied error
+- putting retention on object in bucket without object lock enabled returns
+InvalidRequest error
+- putting valid retention period updates object metadata
+
+### getObjectRetention
+
+- getting retention as non-bucket-owner user returns AccessDenied error
+- getting retention when none is set returns NoSuchObjectLockConfiguration
+error
+- getting retention returns correct object retention period
+
+### putObjectLegalHold
+
+- putting legal hold as non-bucket-owner user returns AccessDenied error
+- putting legal hold on object in bucket without object lock enabled returns
+InvalidRequest error
+- putting valid legal hold updates object metadata
+
+### getObjectLegalHold
+
+- getting legal hold as non-bucket-owner user returns AccessDenied error
+- getting legal hold when none is set returns NoSuchObjectLockConfiguration
+error
+- getting legal hold returns correct object legal hold
+
+## End to End Tests
+
+### Scenarios
+
+- Create bucket with object lock enabled. Put object. Put object lock
+configuration. Put another object.
+
+    - Ensure object put before configuration does not have retention period set
+    - Ensure object put after configuration does have retention period set
+
+- Create bucket without object lock. Put object. Enable object lock with token
+and put object lock configuration. Put another object.
```

Comment:
Are we going to make it possible to be enabled by `token`? In [here](https://github.com/scality/cloudserver/pull/2467/files#diff-f5bc3c260b0f1e4f3bd8617dd2c3a292R187) it suggests that ` we will use a tool to enable object lock by modifying the bucket's configuration` (I'm not sure about what kind of tool though :))